app/controllers/nsm/imports_controller.rb

```diff
@@ -78,6 +108,17 @@ def xml_file
       @xml_file ||= Nokogiri::XML::Document.parse(@form_object.file_upload.tempfile.read, &:noblanks)
     end
 
+    # Extract version from XML
```

Comment:
`version` is already part of the XML file, included in the `<?xml` doctype tag and available in the `.version` method. The point of this ticket as I said was to just enforce that that's included and if it's not we throw an error before we try to validate (since we can't find the right version to validate without the right version) rather than introducing a new field.

This comment applies to all the added-field-related work

Reply:
The XML declaration (<?xml version="1.0"?>) indicates the XML standard version, The XML Declaration provides the XML Parser with version of the XML specification to use encoding. UTF-8, UTF-16, ISO-8859-1 etc. This can't/ shouldn't be used for our versioning.

Reply:
I see, makes sense then. Annoying that we have to add a field for this as it'll constitute _another_ schema push-out.

In that case, the comment against it should be expanded a bit to clarify the purpose of the field, maybe

> The version of the schema to compare an export against.

and an additional `xs:documentation` to clarify that we expect an integer, not to pass a SemVer number or anything (minOccurs/maxOccurs also isn't needed for an integer, if they try and submit a self-closing or empty tag, it throws an error as you can't have "nothing" for a number)